plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -233,4 +245,79 @@ public static function get_note_from_db( $note_from_db ) {
 
 		return $note_from_db;
 	}
+
+	/**
+	 * Add the debug tools to the WooCommerce debug tools (WooCommerce > Status > Tools).
+	 *
+	 * @param array $tools a list of tools.
+	 *
+	 * @return mixed
+	 */
+	public static function add_debug_tools( $tools ) {
+		$tools['refres_remote_inbox_notifications'] = array(
+			'name'     => __( 'Refresh Remote Inbox Notifications', 'woocommerce' ),
+			'button'   => __( 'Refresh', 'woocommerce' ),
+			'desc'     => __( 'This will refresh the remote inbox notifications', 'woocommerce' ),
+			'callback' => function () {
+				RemoteInboxNotificationsDataSourcePoller::get_instance()->read_specs_from_data_sources();
+				RemoteInboxNotificationsEngine::run();
+
+				return __( 'Remote inbox notifications have been refreshed', 'woocommerce' );
+			},
+		);
+
+		$tools['delete_remote_notification'] = array(
+			'name'     => __( 'Delete Remote Notification', 'woocommerce' ),
```

Comment:
Technically speaking, this tool can be used for all inbox notes, not just Remote Notifications. Should we name it as such?

Reply:
@ilyasfoo Good catch. I had no idea other notes were saved into the notes table. I'll have to test it out. I'll make necessary changes soon.

Reply:
@ilyasfoo I looked at the code to confirm, but I think only Remote Inbox Notifications are saved into `wp_wc_admin_notes`. Were you suggesting to support other notifications?

Reply:
@moon0326  I was able to find `wc-admin-wc-helper-connection`, which I'm pretty sure is from the [code here](https://github.com/woocommerce/woocommerce/blob/60c4807e4b60d0640c838a1c0a0568693fec57bb/plugins/woocommerce/src/Internal/Admin/Notes/WooSubscriptionsNotes.php#L21), unless WCCOM also has it? 🤔 

<img width="588" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/b9b90ca7-362f-4d56-8b61-1b5834542d67">

Reply:
@ilyasfooAh, I see what you mean. As far as I know, we no longer add class-based notes. Unless I'm mistaken, that is also a remote inbox notification and it should be searchable.

Reply:
> As far as I know, we no longer add class-based notes

@moon0326  Not sure if I understood you correctly, but I can still get this inbox note shown in a JN with this branch (the note is changed very recently in [this PR](https://github.com/woocommerce/woocommerce/pull/48128/files#diff-7c49164f739ff06a4ba887854edb992f0ec80d5d6a2a627e240caaa4a292e528L200-L215)):

<img width="764" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/35a345df-7a96-4804-9bb9-b72249faf8e8">

> Unless I'm mistaken, that is also a remote inbox notification and it should be searchable.

I also could not find `wc-admin-wc-helper-connection` in the [remote inbox JSON](https://woocommerce.com/wp-json/wccom/inbox-notifications/2.0/notifications.json?locale=en_US)

Reply:
@ilyasfoo Ah I think I see what you mean now. 

>  this tool can be used for all inbox notes, not just Remote Notifications

I always considered anything added to the notes (including [this class](https://github.com/woocommerce/woocommerce/blob/60c4807e4b60d0640c838a1c0a0568693fec57bb/plugins/woocommerce/src/Internal/Admin/Notes/WooSubscriptionsNotes.php#L21)) table a remote inbox notification because the table is only used by remote inbox notification system. I can see that we can call them differently.

Do you think "Inbox Notification" makes more sense in this case? Do you have a suggestion for the name?

Reply:
> Do you think "Inbox Notification" or "Inbox Note" makes more sense in this case? or do you have a different suggestion?

`Inbox Notification` is my best choice, thanks!

Reply:
@ilyasfoo Good choice!

Reply:
@ilyasfoo Updated in https://github.com/woocommerce/woocommerce/pull/48961/commits/8abe0323c5939bbbe0898156975cdded81cec6b1

Reply:
@ilyasfoo That's a good catch. What do you think we also refresh class-based notes and call the tool Inbox notification? 

I think we can run the function from https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Events.php#L143-L145

Reply:
Looks great to me! Thanks, @moon0326!